### PR TITLE
Update spec: make top topics selection UA specific

### PR DESCRIPTION
**Pull request by xyaoinum**
Allow the browser to use their own algorithm

***
<a href="https://pr-preview.s3.amazonaws.com/xyaoinum/topics/pull/203.html" title="Last updated on Jun 22, 2023, 4:02 PM UTC (a34a2d2)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/patcg-individual-drafts/topics/203/104785a...xyaoinum:a34a2d2.html" title="Last updated on Jun 22, 2023, 4:02 PM UTC (a34a2d2)">Diff</a>

### CODE REVIEW

**Comment by xyaoinum** [comment]
@domenic PTAL. Thanks!

File: spec.bs [hunk 1]
```diff
@@ -227,6 +227,32 @@ spec: html; urlPrefix: https://www.rfc-editor.org/rfc/
         1. [=set/Append=] |callerOrigin| to |topicsHistoryEntry|'s [=topics caller origins=].
   </div>
 
+  <h2 id="derive-top-5-topics-header">Derive top 5 topics</h2>
+
+  Given a [=list=] of [=topics history entries=] <var ignore=''>historyEntriesForUserTopics</var>, the browser should provide an algorithm to <dfn>derive top 5 topics</dfn>, that are believed to be valuable for the Topics callers.
```
Comment (domenic) [inline spec.bs:232]: You should state the algorithm's expected return type. I think it's a list of 5 topic IDs?

File: spec.bs [hunk 2]
```diff
@@ -227,6 +227,32 @@ spec: html; urlPrefix: https://www.rfc-editor.org/rfc/
...
+
+    <div class="example">
+      <div algorithm="example-algorithm">
+
+      Given a [=list=] of [=topics history entries=] |historyEntriesForUserTopics|:
+        1. Let |topicsCount| be an empty map.
+        1. For each [=topics history entry=] |historyEntry| in |historyEntriesForUserTopics|:
+            1. [=Classify=] |historyEntry|'s [=topics history entry/topics calculation input data=] into |topicIds|.
+            1. For each |topicId| in |topicIds|:
+              1. If |topicsCount|[|topicId|] does not exist:
```
Comment (domenic) [inline spec.bs:245]: In the output at https://pr-preview.s3.amazonaws.com/xyaoinum/topics/pull/203.html#derive-top-5-topics this indentation got lost. I think it's just missing a space.

File: spec.bs [hunk 3]
```diff
@@ -227,6 +227,32 @@ spec: html; urlPrefix: https://www.rfc-editor.org/rfc/
...
+                  1. Initialize |topicsCount|[|topicId|] to 0.
+              1. Increment |topicsCount|[|topicId|] by 1.
+        1. Let |top5Topics| be a list containing the top up to 5 |topicId|s in |topicsCount|'s keys set, where the |topicId|s with more count are retrieved first.
```
Comment (domenic) [inline spec.bs:248]:
```suggestion
        1. Let |top5Topics| be a list containing the top up to 5 |topicId|s in |topicsCount|'s [=map/keys=], where the |topicId|s with more count are retrieved first.
```

File: spec.bs [hunk 4]
```diff
@@ -227,6 +227,32 @@ spec: html; urlPrefix: https://www.rfc-editor.org/rfc/
...
+  <div class="note">
+  Chrome's initial release scores topics by the frequency of page loads with that topic.
+
+    <div class="example">
+      <div algorithm="example-algorithm">
+
+      Given a [=list=] of [=topics history entries=] |historyEntriesForUserTopics|:
+        1. Let |topicsCount| be an empty map.
+        1. For each [=topics history entry=] |historyEntry| in |historyEntriesForUserTopics|:
+            1. [=Classify=] |historyEntry|'s [=topics history entry/topics calculation input data=] into |topicIds|.
```
Comment (domenic) [inline spec.bs:243]: I would phrase this as "Let |topicIds| be the result of [classifying] ..."

**Review by domenic — APPROVED** [review]
A few nits, and stating the return type is important. But looks good otherwise.

**Comment by ratchanonsuttawas** [comment]
> Allow the browser to use their own algorithm
> 
> [Preview](https://pr-preview.s3.amazonaws.com/xyaoinum/topics/pull/203.html) | [Diff](https://pr-preview.s3.amazonaws.com/patcg-individual-drafts/topics/203/104785a...xyaoinum:a34a2d2.html)

**Pull request merged**